.github/workflows/test.yml

```
@@ -0,0 +1,357 @@
+        # Use PostgreSQL on Linux, SQLite on Windows/macOS
+        DATABASE_URL: ${{ matrix.os == 'ubuntu-latest' && 'postgresql://testuser:testpassword@localhost:5432/testdb' || 'sqlite:///./test.db' }}
+        TEST_DATABASE_URL: ${{ matrix.os == 'ubuntu-latest' && 'postgresql+asyncpg://testuser:testpassword@localhost:5432/testdb' || 'sqlite+aiosqlite:///./test.db' }}
+        ENVIRONMENT: testing
+      run: |
+        uv run pytest tests/integration/ \
+          --verbose \
+          --tb=short \
+          --junit-xml=junit-integration.xml
+    
+    - name: 📊 Upload Coverage to Codecov
+      uses: codecov/codecov-action@v3
```

actionlint (1.7.7)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)